src/vault/entities/key.go

```diff
@@ -8,4 +10,7 @@ type Key struct {
 	PublicKey  string            `json:"publicKey"`
 	Namespace  string            `json:"namespace,omitempty"`
 	Tags       map[string]string `json:"tags,omitempty"`
+	CreatedAt  time.Time         `json:"createdAt"`
+	UpdatedAt  time.Time         `json:"updatedAt"`
+	Version    int               `json:"version"`
```

Comment:
Dont we want a version in string? As what is implemented for secret

Reply:
version is `int` in Hashicorp so I used the same. It is string in AKV, for me either is fine. But if we use string, then we don't have chronological ordering and we need to use a random string generator.